### PR TITLE
PRL: Document noise function eligibility warnings

### PR DESCRIPTION
**Pull request by NathanielBlairStahn**
JIRA ticket: [SSCI-1397](https://jira.ihme.washington.edu/browse/SSCI-1397)

This PR documents how to deal with the row eligibility requirements (including when and how to raise user warnings) in three different noise functions: Nicknames; Copy from within household; and Guardian-based duplication. This behavior was already implemented (based on Slack conversations) for Nicknames, and was decided on but not yet implemented for the other two noise functions. The application of these noise functions to the correct fraction of eligible rows is complicated by our shard-based distributed processing approach, so we decided to document more of the implementation details than we usually provide the engineers.

Also, I added a to-do to add more details about how we implemented Copy from within household noise, as what's there seems like an incomplete specification.

### CODE REVIEW

File: docs/source/models/concept_models/vivarium_census_synthdata/concept_model.rst [hunk 1]
```diff
@@ -3670,7 +3708,73 @@ dataset. Select the guardian at random.
...
+    row-based noise). If :math:`p>F`, raise a warning to the user before
+    noising starts.
+
+#.  When noising each shard, compute the fraction :math:`f` of rows eligible
+    for noise in the shard, and apply the noise to each eligible row in the
+    shard with probability :math:`\min\{p/f, 1\}`. Do **not** raise a warning to
+    the user for every shard with :math:`p>f`.
+
+This strategy has a known limitation: It systematically adds less noise to the
+dataset than requested when :math:`p` sufficiently large. This is obvious when
```
Comment (zmbc) [inline docs/source/models/concept_models/vivarium_census_synthdata/concept_model.rst:3742]:
```suggestion
dataset than requested when :math:`p` is sufficiently large. This is obvious when
```

File: docs/source/models/concept_models/vivarium_census_synthdata/concept_model.rst [hunk 2]
```diff
@@ -3670,7 +3708,73 @@ dataset. Select the guardian at random.
...
+
+We are willing to accept the above limitation for simplicity's sake. Moreover,
+the impact of this issue should be mitigated when we change our shard-based
+approach to data storage so that the different data files have similar sizes
+rather than some datasets having tiny shards like ACS currently does.
+
+Another potentially ambiguous point in the above strategy is which data to use
+to pre-compute the eligible fraction :math:`F`. Since these pre-computed values
+are merely for user warnings, the researchers are flexible as to how closely the
+:math:`F` values should match match the corresponding datasets, and we'll let
```
Comment (zmbc) [inline docs/source/models/concept_models/vivarium_census_synthdata/concept_model.rst:3763]:
```suggestion
:math:`F` values should match the corresponding datasets, and we'll let
```

File: docs/source/models/concept_models/vivarium_census_synthdata/concept_model.rst [hunk 3]
```diff
@@ -3670,7 +3708,73 @@ dataset. Select the guardian at random.
...
+the engineers decide on the exact implementation. Here are a couple options: If
+the pre-computed values are stored in a separate metadata file generated with
+each simulation run, then :math:`F` can be computed for each applicable
+(dataset, year, noise type) combination that can be requested by the user. On
+the other hand, if it's easier to store a single value of :math:`F` for each
+(dataset, noise type) without varying across years, then we should should use
+the dataset for the year 2030; since 2030 is the midpoint of our simulation, we
+expect it to be sufficiently representative of the data across all simulated
+years. This level of imprecision is sufficient for the user warning, which
```
Comment (zmbc) [inline docs/source/models/concept_models/vivarium_census_synthdata/concept_model.rst:3772]:
```suggestion
years.
If even this is too complicated, a single :math:`F` can be computed for a noise type, and not vary between datasets;
the decennial census dataset should be used to do this.
The level of imprecision is acceptable for the user warning, which
```

Isn't this what was actually implemented for nicknames? I'm pretty sure they used just the 2030 decennial.

**Review by zmbc — APPROVED** [review]
Looks great. Thanks so much for getting this actually documented instead of stored in our heads and in Slack messages!

File: docs/source/models/concept_models/vivarium_census_synthdata/concept_model.rst (hunk 3 above)
Comment (NathanielBlairStahn) [reply docs/source/models/concept_models/vivarium_census_synthdata/concept_model.rst:3772]: Oh yeah, I think you're right -- I forgot the details of what was actually implemented.

**Review by aflaxman — APPROVED** [review]

**Review by pletale — APPROVED** [review]

**Pull request merged**